src/fondant/component_spec.py

```diff
@@ -47,7 +47,8 @@ def python_type(self) -> t.Any:
             "dict": json.loads,
             "list": json.loads,
         }
-        return lookup[self.type]
+        map_fn = lookup[self.type]
+        return lambda value: map_fn(value) if value != "None" else None  # type: ignore
```

Comment:
This can be a bit confusing so i'll recap a bit:

* In kubeflow optional types that default to `None` are defined as optional in the spec with no `constant` runtime value. See `input3` [here](https://github.com/kubeflow/pipelines/blob/cfe671c485d4ee8514290ee81ca2785e8bda5c9b/sdk/python/test_data/pipelines/component_with_optional_inputs.py) and [here](https://github.com/kubeflow/pipelines/blob/cfe671c485d4ee8514290ee81ca2785e8bda5c9b/sdk/python/test_data/pipelines/component_with_optional_inputs.yaml). They also should not be passed to the componentOp so that's why we remove them [here](https://github.com/ml6team/fondant/blob/785940eece05cea7edc58c76033cf2604aa8db66/src/fondant/compiler.py#L449)
* In docker, all arguments are passed as strings. Above we're defining a map function above a type that converts back to their original type. In case that value is a `None` string, we're converting it back to a `Nonetype`. We actually has this in the v1 implementation, not sure why it was removed

Reply:
Ok, I was a bit confused by the notation, but looking at it again, it's clear to me why this is needed. Thanks!